packages/datadog-instrumentations/src/fastify.js

```diff
@@ -154,9 +155,12 @@ function preParsing (request, reply, payload, done) {
 }
 
 function wrapSend (send, req) {
-  return function sendWithTrace (error) {
-    if (error instanceof Error) {
-      errorChannel.publish({ req, error })
+  return function sendWithTrace (payload) {
+    if (payload instanceof Error) {
+      errorChannel.publish({ req, error: payload })
+    } else if (responseJsonReadCh.hasSubscribers) {
+      const res = getRes(this)
+      responseJsonReadCh.publish({ req, res, body: payload })
```

Comment:
`reply.send` expect a lot of different types: `Error`, `string`, `Buffer`, `Stream`/`ReadableStream`, TypedArrays, and other objects. We are only interested in other objects, and the subscriber shouldn't understand about different options that this method support. We should discar, at least, all the `Buffer`, `Stream`/`ReadableStream`, TypedArrays and send only when we have the real and final object.

Reply:
Here the link to the docs: https://fastify.dev/docs/latest/Reference/Reply/#senddata